app/src/layouts/kanban/index.ts

```diff
@@ -0,0 +1,596 @@
+				} else {
+					const related = relationsStore.relations.find(
+						(relation) =>
+							relation.collection === props.collection &&
+							relation.field === field.field &&
+							relation.related_collection === 'directus_users'
+					);
+
+					return related !== undefined;
+				}
+			},
+			group: (field) => {
+				if (field.meta?.options && Object.keys(field.meta?.options).includes('choices')) {
+					return Object.keys(field.meta?.options).includes('choices');
+				}
+
+				const relation = relationsStore.relations.find(
+					(relation) => getRelationType({ relation, collection: collection.value, field: field.field }) === 'm2o'
+				);
+
+				return !!relation;
+			},
+
+			file: (field) => {
+				if (field.field === '$thumbnail') return true;
+
+				const relation = relationsStore.relations.find((relation) => {
+					return (
+						relation.collection === props.collection &&
+						relation.field === field.field &&
+						relation.related_collection === 'directus_files'
+					);
+				});
+
+				return !!relation;
+			},
+		});
+
+		const {
+			groupField,
+			groupTitle,
+			imageSource,
+			titleField,
+			textField,
+			crop,
+			selectedGroup,
+			dateField,
+			tagsField,
+			userField,
+			showUngrouped,
+			userFieldJunction,
+			userFieldType,
+		} = useLayoutOptions();
+
+		const {
+			groups,
+			groupsSortField,
+			groupsPrimaryKeyField,
+			groupTitleFields,
+			groupsCollection,
+			changeGroupSort,
+			addGroup,
+			editGroup,
+			deleteGroup,
+			isRelational,
+		} = useGrouping();
+
+		const { items, loading, error, totalPages, itemCount, totalCount, changeManualSort } = useItems(collection, {
+			sort,
+			limit,
+			page,
+			fields,
+			filter,
+			search,
+		});
+
+		const limitWarning = computed(() => items.value.length >= limit.value);
+
+		const groupedItems = computed<Group[]>(() => {
+			const groupsCollectionPrimaryKeyField = groupsPrimaryKeyField.value?.field;
+			const groupTitleField = groupTitle?.value || groupsCollectionPrimaryKeyField;
+			const group = groupField.value;
+			const pkField = primaryKeyField.value?.field;
+			const itemGroups: Record<string | number, Group> = {};
+
+			if (!pkField || !group) return [];
+
+			if (isRelational.value && !groupTitleField) return [];
+
+			groups.value.forEach((group, index) => {
+				const id =
+					isRelational.value && groupsCollectionPrimaryKeyField ? group[groupsCollectionPrimaryKeyField] : group.value;
+
+				const title = String(isRelational.value && groupTitleField ? group[groupTitleField] : group.text);
+
+				itemGroups[id] = {
+					id,
+					title: translate(title),
+					items: [],
+					sort: index,
+				};
+			});
+
+			if (showUngrouped.value) {
+				itemGroups['_ungrouped'] = {
+					id: null,
+					items: [],
+					title: '_ungrouped',
+					sort: -1,
+				};
+			}
+
+			items.value.forEach((item, index) => {
+				if (!group || !pkField) return;
+
+				const junctionField = userFieldJunction.value?.meta?.junction_field ?? undefined;
```

Comment:
```suggestion
				const junctionField = userFieldJunction.value?.meta?.junction_field;
```
optional chaining already falls back to `undefined`